src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PotentialEditorConfigDataSource.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    [Export(typeof(IFileWatchProviderDataSource))]
+    internal class PotentialEditorConfigDataSource : ProjectValueDataSourceBase<IFileWatchProvider>, IFileWatchProviderDataSource
+    {
+        private readonly ConfiguredProject _project;
+
+        private int _version;
+        private IDisposable _projectRuleSourceLink;
+
+        /// <summary>
+        /// The block that receives updates from the active tree provider.
+        /// </summary>
+        private IBroadcastBlock<IProjectVersionedValue<IFileWatchProvider>> _broadcastBlock;
+
+        /// <summary>
+        /// The public facade for the broadcast block.
+        /// </summary>
+        private IReceivableSourceBlock<IProjectVersionedValue<IFileWatchProvider>> _publicBlock;
+
+        [ImportingConstructor]
+        public PotentialEditorConfigDataSource(ConfiguredProject project)
+            : base(project.Services)
```

Comment:
synchronousDisposable:true. You have are taking no locks in Dispose.

Reply:
Done.